features/dd-sdk-android-trace-otel/src/main/kotlin/com/datadog/android/trace/opentelemetry/OtelTracerProvider.kt

```diff
@@ -127,43 +134,51 @@ class OtelTracerProvider(
         /**
          * Builds a [TracerProvider] based on the current state of this Builder.
          */
-        fun build(): OtelTracerProvider {
-            val tracingFeature = sdkCore.getFeature(Feature.TRACING_FEATURE_NAME)
-                ?.unwrap<TracingFeature>()
-            if (tracingFeature == null) {
-                sdkCore.internalLogger.log(
-                    InternalLogger.Level.ERROR,
-                    InternalLogger.Target.USER,
-                    { TRACING_NOT_ENABLED_ERROR_MESSAGE }
-                )
-            } else {
-                sdkCore.updateFeatureContext(Feature.TRACING_FEATURE_NAME) {
-                    it[TracingFeature.IS_OPENTELEMETRY_ENABLED_CONFIG_KEY] = true
-                    it[TracingFeature.OPENTELEMETRY_API_VERSION_CONFIG_KEY] = BuildConfig.OPENTELEMETRY_API_VERSION_NAME
-                }
-            }
-            val coreTracer = CoreTracer.CoreTracerBuilder(sdkCore.internalLogger)
-                .withProperties(properties())
-                .serviceName(serviceName)
-                .writer(tracingFeature?.otelDataWriter ?: NoOpOtelWriter())
-                .partialFlushMinSpans(partialFlushThreshold)
-                .idGenerationStrategy(IdGenerationStrategy.fromName("SECURE_RANDOM", false))
-                .build()
-            coreTracer.addScopeListener(object : ScopeListener {
-                override fun afterScopeActivated() {
-                    val activeSpanContext = coreTracer.activeSpan()?.context()
-                    if (activeSpanContext != null) {
-                        val activeSpanId = activeSpanContext.spanId.toString()
-                        val activeTraceId = activeSpanContext.traceId.toString()
-                        sdkCore.addActiveTraceToContext(activeTraceId, activeSpanId)
+        fun build(): TracerProvider {
+            return executeSafelyOnAndroid23AndBelow(
```

Comment:
nit: current method name gives an impression that code will be executed only on API 23 and below, which is not true. maybe it is worth to rename it to `executeIfAboveAndroid24`?

Reply:
hmm, but that name suggestion will mean that it will only be executed on `Android24` and above for me :). What if I just call it `executeSafely` ?

Reply:
Normally the method name should reflect when lambda will be executed, not when it won't.

Reply:
yes definitely and actually the name your proposed it's fine, my mind is a bit blurry after all this nordic weather :)